pkg/apis/serving/v1beta1/inference_service_validation.go

```diff
@@ -141,21 +142,19 @@ func validateInferenceServiceAutoscaler(isvc *InferenceService) error {
 				switch class {
 				case constants.AutoscalerClassHPA:
 					if metric, ok := annotations[constants.AutoscalerMetrics]; ok {
-						return validateHPAMetrics(ScaleMetric(metric))
+						return class, validateHPAMetrics(ScaleMetric(metric))
 					} else {
-						return nil
+						return class, nil
 					}
-				case constants.AutoscalerClassExternal:
-					return nil
-				default:
-					return fmt.Errorf("unknown autoscaler class [%s]", class)
+				case constants.AutoscalerClassExternal, constants.AutoscalerClassNone:
+					return class, nil
 				}
 			}
 		}
-		return fmt.Errorf("[%s] is not a supported autoscaler class type", value)
+		return class, fmt.Errorf("[%s] is not a supported autoscaler class type", value)
```

Comment:
why return a non-supported autoscaler class?

Reply:
Good question... I don't have a good answer.
I'm changing it to return blank.